base/array.jl

```diff
@@ -653,6 +653,11 @@ function lexcmp(a::Array{UInt8,1}, b::Array{UInt8,1})
     return c < 0 ? -1 : c > 0 ? +1 : cmp(length(a),length(b))
 end
 
+# use memcmp for == on integer types
+=={T<:Union{Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64}}(a::Array{T,1}, b::Array{T,1}) =
+    (len = length(a)) == length(b) &&
+    ccall(:memcmp, Int32, (Ptr{T}, Ptr{T}, UInt), a, b, len * sizeof(T)) == 0
```

Comment:
I realize you're copy pasting most of the form of this, but I think it would look clearer as
```
function =={T<:BitInteger}(a::Array{T,1}, b::Array{T,1})
    len = length(a)
    return len == length(b) &&
        0 == ccall(:memcmp, Int32, (Ptr{T}, Ptr{T}, Csize_t), a, b, len * sizeof(T))
end
```